src/interface/blas2_interface.hpp

```diff
@@ -678,19 +779,37 @@ typename sb_handle_t::event_t _tbsv_impl(sb_handle_t& sb_handle, index_t _N,
                  : ((roundUp<index_t>(_N, subgroup_size) / subgroup_size) - 1);
   sync_vec[1] = sync_vec[0];
 
-  auto sync_buffer =
-      blas::make_sycl_iterator_buffer<int32_t>(sync_vec, sync_vec.size());
-  auto sync = make_vector_view(sync_buffer, one::value(), sync_vec.size());
+  constexpr bool is_usm = std::is_pointer<container_t0>::value;
+  auto queue = sb_handle.get_queue();
+
+  auto sync_buffer = blas::helper::allocate < is_usm
+                         ? blas::helper::AllocType::usm
+                         : blas::helper::AllocType::buffer,
+       index_t > (sync_vec.size(), queue);
```

Comment:
```suggestion
       int32_t > (sync_vec.size(), queue);
```

Reply:
I would strongly suggest not to hardcode index type. Intel in onemkl uses `int64_t` for the index, so we need to let the index type to be adopted to what has been chosen through template

Reply:
As you prefer, @mehdi-goli.

But, not all the devices support 64 bit atomic operations. Moreover, specifying `int32_t` will guarantee that this will have 32 bits no matter what host platform we are using.

Reply:
If the device does not support we can set the index through template to be 32. But when we put this code as a backend of onemkl intel all the index must be 64. The point is when we have it in index we can control it for all the devices. But in onemkl case that can lead to index overflow as they are using very large index hence they force it to be 64

Reply:
This and other suggestions below have been addressed in https://github.com/codeplaysoftware/sycl-blas/pull/441

Reply:
We decided to stick to `int32_t` for the above.